### DISCUSSION TITLE
Unclarity about multi channels setup

### DISCUSSION THREAD

**Discussion by IgorKhomenko** [saleor/saleor#9112] [Deployment]
Hey everyone, 

I just came up with one unclarity about multi channels setup 
https://docs.saleor.io/docs/3.0/developer/channels

From dashboard perspective, it's all clear: we can create diff channels and assign diff products set to it. 

But, from a storefront perspective, I found we have to hardcode the channel which this storefront will be using https://github.com/saleor/react-storefront/blob/main/.env#L2 , which directs me to a set of questions:

1) if I want to have 2 channels, e.g. 2 diff countries or web and mobile, should I setup and run 2 diff storefronts with diff channel env? 
2) if yes, then how I can make it look seamless for end users? Should I setup some AWS Route53 balancing strategy based on geolocation or user agent and connect 2 storefronts behind it? 

Is my understanding correct here? 

Thank you

**Comment by krzysztofwolski** [comment]
It's all depends on what would you like to achieve.

Example 1: separate consumer and business 
Channels:
- consumer
- business

If you would like to have a separate business catalog, you can deploy 2 separate storefronts. One for b2b and one for consumers.

Example 2: multi-region channels
Channels:
- eu-region
- usa-region

That's the situation we have in react storefront. Based on user selection (or URL), react application pulls data from the related channel. `default-channel` is used as the fallback if we don't have enough data. 
If Mobile applications use the same catalog, the same channels can be used.
I would create a separate mobile channel only if you have exclusive products or prices in the app.

Your use case seems like example 2, so no multi-deployment is required.

**Comment by IgorKhomenko** [comment]
@krzysztofwolski thanks for the detailed answer 

Initially I did not realise this switch flow. It makes the things more clear now 

![Screenshot 2022-02-08 at 17 31 54](https://user-images.githubusercontent.com/70977170/153020379-3b569b50-041c-422a-94ec-fd7be46b045f.png)
